src/types/ArrayType.ts

```diff
@@ -33,19 +38,23 @@ export class ArrayType implements BscType {
         return false;
     }
 
-    public isConvertibleTo(targetType: BscType) {
-        return this.isAssignableTo(targetType);
+    public isConvertibleTo(targetType: BscType, context?: TypeContext) {
+        return this.isAssignableTo(targetType, context);
     }
 
-    public toString() {
-        return `Array<${this.innerTypes.map((x) => x.toString()).join(' | ')}>`;
+    public toString(context?: TypeContext) {
+        return `Array<${this.innerTypes.map((x) => x.toString(context)).join(' | ')}>`;
```

Comment:
I think this should be changed to `\`${this.defaultType.toString(context)}[]\``  ... thoughts?

Reply:
I had originally imagined this to support arrays with multiple types. For example, something like:
`Array<string|integer>`. I'm fine with your suggestion because I don't know that we're ready for my design quite yet. But would `this.defaultType` already support that concept?